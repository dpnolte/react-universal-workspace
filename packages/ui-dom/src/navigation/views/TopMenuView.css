.top-menu-link-text {
  padding-left: 5px;
  padding-top: 3px;
  color: #bbb;
}
.container {
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  display: flex;
  flex-direction: column;
}

.header {
  height: 60px;
}

.content {
  display: flex;
  height: 100%;
}

/* A simple, css only, (some-what) responsive menu */

nav.top-menu-nav {
  position: relative;
  font-size: 13px;
  text-transform: uppercase;
  text-align: center;
  /* Permalink - use to edit and share this gradient: http://colorzilla.com/gradient-editor/#2b2b2b+7,333333+100 */
  background: #2b2b2b; /* Old browsers */
  background: -moz-linear-gradient(
    top,
    #2b2b2b 7%,
    #333333 100%
  ); /* FF3.6-15 */
  background: -webkit-linear-gradient(
    top,
    #2b2b2b 7%,
    #333333 100%
  ); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(
    to bottom,
    #2b2b2b 7%,
    #333333 100%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#2b2b2b', endColorstr='#333333',GradientType=0 ); /* IE6-9 */
}

.top-menu-nav a {
  text-decoration: none;
  color: #fff;
  display: block;
}

.top-menu-nav ul {
  list-style: none;
  margin-block-start: 0;
  margin-block-end: 0;
  position: relative;
  text-align: left;
}

.top-menu-nav li {
  float: left;
}

/* clear'n floats */
.top-menu-nav ul:after {
  clear: both;
}

.top-menu-nav ul:before,
.top-menu-nav ul:after {
  content: ' ';
  display: table;
}

/* prime */
.top-menu-nav ul.primary li a {
  padding: 20px 30px;
  border-right: 1px solid #3d3d3d;
  display: flex;
  flex-direction: row;
}

.top-menu-nav ul.primary li a:hover {
  color: #000;
}

/* subs */
.top-menu-nav ul.sub {
  position: absolute;
  z-index: 200;
  box-shadow: 2px 2px 0 #bebebe;
  width: 20%;
  display: none;
}

.top-menu-nav ul.sub li {
  float: none;
  margin: 0;
}

.top-menu-nav ul.sub li a {
  border-bottom: 1px dotted #ccc;
  border-right: none;
  color: #000;
  padding: 15px 30px;
}

.top-menu-nav ul.sub li:last-child a {
  border-bottom: none;
}

.top-menu-nav ul.sub li a:hover {
  color: #000;
  background: #eeeeee;
}

/* sub display*/
.top-menu-nav ul.primary li:hover ul {
  display: block;
  background: #fff;
  margin-top: -3px;
}

/* keeps the tab background white */
.top-menu-nav ul.primary li:hover a {
  background: #fff;
  color: #666;
  text-shadow: none;
}

.top-menu-nav ul.primary li:hover > a .top-menu-link-text {
  color: #999;
}
