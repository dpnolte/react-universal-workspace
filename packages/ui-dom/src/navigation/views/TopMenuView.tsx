/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react'
import {
  SceneView,
  NavigationScreenProp,
  NavigationRoute,
  NavigationDescriptor,
} from '@react-navigation/core'
import { Link } from '@react-navigation/web'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import './TopMenuView.css'

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface INavigationParams {}
interface IProps {
  descriptors: { [key: string]: NavigationDescriptor }
  navigation: NavigationScreenProp<
    NavigationRoute<INavigationParams>,
    INavigationParams
  >
  navigationConfig: {}
}

export const TopMenuView = (props: IProps) => {
  const { descriptors, navigation } = props
  const activeKey = navigation.state.routes[navigation.state.index].key
  const descriptor = descriptors[activeKey]
  return (
    <div className="container">
      <header className="header">
        <nav className="top-menu-nav">
          <ul className="primary">
            <li>
              <Link
                routeName="Home"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.HomeLink,
                }}
              >
                <Icon name="home" size={20} color="#bbb" />
                <span className="top-menu-link-text">Home</span>
              </Link>
            </li>
            <li>
              <Link
                routeName="Docs"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.DocsLink,
                }}
              >
                <Icon name="file-document-outline" size={20} color="#bbb" />
                <span className="top-menu-link-text">Documents</span>
              </Link>
              {/* <ul className="sub">
                <li>
                  <Link
                    routeName="Docs"
                    navigation={navigation}
                    anchorProps={{
                      'data-testid': TopMenuViewTestIds.DocsLink,
                    }}
                  >
                    Doc 1
                  </Link>
                </li>
                  </ul> */}
            </li>
            <li>
              <Link
                routeName="TodoList"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.TodoListLink,
                }}
              >
                <Icon name="format-list-checkbox" size={20} color="#bbb" />
                <span className="top-menu-link-text">Todo List</span>
              </Link>
            </li>
            <li>
              <Link
                routeName="About"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.AboutLink,
                }}
              >
                <Icon name="vector-arrange-above" size={20} color="#bbb" />
                <span className="top-menu-link-text">About</span>
              </Link>
            </li>
          </ul>
        </nav>
      </header>
      <section className="content">
        <SceneView
          navigation={descriptor.navigation}
          component={descriptor.getComponent()}
        />
      </section>
    </div>
  )
}

export const TopMenuViewTestIds = {
  HomeLink: 'top-menu-home-link',
  DocsLink: 'top-menu-docs-link',
  AboutLink: 'top-menu-about-link',
  TodoListLink: 'top-menu-todo-list-link',
}
