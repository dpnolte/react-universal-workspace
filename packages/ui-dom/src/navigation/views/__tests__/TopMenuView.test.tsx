import React from 'react'
import renderer from 'react-test-renderer'
import { TopMenuView } from '../TopMenuView'

describe('TopMenuView unit tests', () => {
  let mockedNavigation
  let mockedDescriptors: { Test: { getComponent: jest.Mock<any, any> } }
  let mockedProps: { navigation: any; descriptors: any; navigationConfig: {} }
  beforeEach(() => {
    mockedNavigation = {
      state: {
        params: {},
        routes: [
          {
            key: 'Test',
          },
        ],
        index: 0,
      },
      router: {
        getStateForAction: jest.fn(),
        getPathAndParamsForState: () => {
          return { path: 'Test', params: {} }
        },
      },
      dispatch: jest.fn(),
      goBack: jest.fn(),
      dismiss: jest.fn(),
      navigate: jest.fn(),
      openDrawer: jest.fn(),
      closeDrawer: jest.fn(),
      toggleDrawer: jest.fn(),
      getParam: jest.fn(),
      setParams: jest.fn(),
      addListener: jest.fn(),
      push: jest.fn(),
      replace: jest.fn(),
      pop: jest.fn(),
      popToTop: jest.fn(),
      isFocused: jest.fn(),
      dangerouslyGetParent: jest.fn(),
    }
    mockedDescriptors = {
      Test: {
        getComponent: jest.fn(() => () => <div>scene view content</div>),
      },
    }
    mockedProps = {
      navigation: mockedNavigation,
      descriptors: mockedDescriptors,
      navigationConfig: {},
    }
  })

  it('should match snapshot', () => {
    const rendered = renderer.create(<TopMenuView {...mockedProps} />).toJSON()

    expect(rendered).toMatchSnapshot()
  })
})
