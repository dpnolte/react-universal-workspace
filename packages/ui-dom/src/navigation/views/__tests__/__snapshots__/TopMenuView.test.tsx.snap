// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenuView unit tests should match snapshot 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <nav
      className="top-menu-nav"
    >
      <ul
        className="primary"
      >
        <li>
          <a
            data-testid="top-menu-home-link"
            href="/Test"
            onClick={[Function]}
          >
            <div
              className="css-text-76zvg2"
              dir="auto"
              style={
                Object {
                  "color": "rgba(187,187,187,1.00)",
                  "fontFamily": "MaterialCommunityIcons",
                  "fontSize": "20px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                }
              }
            >
              
            </div>
            <span
              className="top-menu-link-text"
            >
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            data-testid="top-menu-docs-link"
            href="/Test"
            onClick={[Function]}
          >
            <div
              className="css-text-76zvg2"
              dir="auto"
              style={
                Object {
                  "color": "rgba(187,187,187,1.00)",
                  "fontFamily": "MaterialCommunityIcons",
                  "fontSize": "20px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                }
              }
            >
              吝
            </div>
            <span
              className="top-menu-link-text"
            >
              Documents
            </span>
          </a>
          <ul
            className="sub"
          >
            <li>
              <a
                data-testid="top-menu-docs-link"
                href="/Test"
                onClick={[Function]}
              >
                Doc 1
              </a>
            </li>
          </ul>
        </li>
        <li>
          <a
            data-testid="top-menu-about-link"
            href="/Test"
            onClick={[Function]}
          >
            <div
              className="css-text-76zvg2"
              dir="auto"
              style={
                Object {
                  "color": "rgba(187,187,187,1.00)",
                  "fontFamily": "MaterialCommunityIcons",
                  "fontSize": "20px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                }
              }
            >
              
            </div>
            <span
              className="top-menu-link-text"
            >
              About
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <section
    className="content"
  >
    <div>
      scene view content
    </div>
  </section>
</div>
`;
